File: JAVA/2244. Minimum Rounds to Complete All Tasks.java
import java.util.*;

class Solution {
    public int minimumRounds(int[] tasks) {
        Map<Integer, Integer> map = new HashMap<>();
        int count = 0;
        for (int task : tasks)
            map.put(task, map.getOrDefault(task, 0) + 1);
        for (int freq : map.values()) {
            while (freq != 0) {
                if (freq == 1)
                    return -1;
                else if (freq == 4) {
                    count += 2;
                    freq -= 4;
                } else if (freq == 2) {
                    count++;
                    freq -= 2;
                } else {
                    count++;
                    freq -= 3;
                }
            }
        }
        return count;
    }
}

File: C++/2244. Minimum Rounds to Complete All Tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& t) {
        map<int,int>m;
        int c=0;
        for(int i:t)
        m[i]++;
        for(auto i=m.begin();i!=m.end();i++){
            while(i->second!=0){
                if(i->second==1)
                return -1;
                else if(i->second==4){
                c+=2;
                i->second-=4;
                }
                else if(i->second==2){
                c++;
                i->second-=2;
                }
                else{
                    c++;
                    i->second-=3;
                }
            }
        }
        return c;
    }
};